### PR TITLE
[JBPM-10071] WIH's abortWorkItem method should be called when aborting workItem through kie-server REST

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-10071

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins run fdb

**Comment by gmunozfe** [comment]
Jenkins run fdb

**Review by gmunozfe — CHANGES_REQUESTED** [review]
An infinite loop is created if the implementation invokes the manager.abortWorkItem, like this:

https://github.com/kiegroup/droolsjbpm-integration/blob/751cc6edcf3421f6454dfdae0a1a9cea96002286/kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/filtered-resources/kjars-sources/async-exec-project-101/src/main/java/com/self/exceptionhandbpm/CustomWorkItemHandler.java#L36

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java [hunk 1]
```diff
@@ -195,6 +195,10 @@ public void abortWorkItem( long id ) {
         // work item may have been aborted
         if ( workItemInfo != null ) {
             WorkItem workItem = (WorkItemImpl) internalGetWorkItem( workItemInfo );
+            WorkItemHandler handler = (WorkItemHandler) this.workItemHandlers.get( workItem.getName() );
+            if ( handler != null ) {
+                handler.abortWorkItem( workItem, this );
+            }
```
Comment (gmunozfe) [inline drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java]: You could set the State to ABORTED before invoking `handler.abortWorkItem`, and also, control first if the state is not ABORTED. I've tested and worked like this:

```suggestion
            if (workItem!=null && workItem.getState() == WorkItem.ABORTED) {
            	return;
            }
            workItem.setState( WorkItem.ABORTED );
           
            WorkItemHandler handler = (WorkItemHandler) this.workItemHandlers.get( workItem.getName() );
	    if ( handler != null) {
	        handler.abortWorkItem( workItem, this );
            }
```

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java [hunk 2]
```diff
@@ -195,6 +195,10 @@ public void abortWorkItem( long id ) {
...
             ProcessInstance processInstance = kruntime.getProcessInstance( workItem.getProcessInstanceId() );
             workItem.setState( WorkItem.ABORTED );
```
Comment (gmunozfe) [inline drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java]: Move this line above

Comment (elguardian) [reply drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java]: done

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java (hunk 1 above)
Comment (elguardian) [reply drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManager.java]: done

**Review by gmunozfe — APPROVED** [review]
Looks good to me, outstanding work @elguardian !
I've also added a PR with a test: https://github.com/kiegroup/droolsjbpm-integration/pull/2829

**Pull request merged**